### PR TITLE
Improvements and additions

### PR DESCRIPTION
**Pull request by amerkoleci**
- Fix Veldrid.SDL not compiling due to nativelibrary.csproj not added to solution.
- Constructors and some helpers for VkOffset2D, VkExtent2D and VkRect2D in order to avoid finger break :)
- Add some alias map for VkSampleCountFlagBits, VkImageType and VkImageViewType instead of having _1, _2d etc

### CODE REVIEW

**Comment by mellinoe** [comment]
Thanks for the improvements. Looks good overall.

* It probably sounds like a nit-pick, but could you replace all of your tabs with 4 spaces? The diff here looks quite massive despite just being a few harmless additions.

File: src/vk/VkExtent2D.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+			return obj is VkExtent2D && Equals((VkExtent2D)obj);
+		}
+
+		/// <summary>
+		/// Returns the hash code for this instance.
+		/// </summary>
+		/// <returns>The hash code.</returns>
+		public override int GetHashCode()
+		{
+			unchecked
```
Comment (mellinoe) [inline src/vk/VkExtent2D.cs]: I don't think this is necessary, `unchecked` should be the default.

File: src/vk/VkOffset2D.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+			return obj is VkOffset2D && Equals((VkOffset2D)obj);
+		}
+
+		/// <summary>
+		/// Returns the hash code for this instance.
+		/// </summary>
+		/// <returns>The hash code.</returns>
+		public override int GetHashCode()
+		{
+			unchecked
```
Comment (mellinoe) [inline src/vk/VkOffset2D.cs]: Ditto; should not be necessary.

File: src/vk/VkRect2D.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+			return obj is VkRect2D && Equals((VkRect2D)obj);
+		}
+
+		/// <summary>
+		/// Returns the hash code for this instance.
+		/// </summary>
+		/// <returns>The hash code.</returns>
+		public override int GetHashCode()
+		{
+			unchecked
```
Comment (mellinoe) [inline src/vk/VkRect2D.cs]: Same

File: src/vk/VkExtent2D.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+		/// <summary>
+		/// Determines whether the specified <see cref="VkExtent2D"/> is equal to this instance.
+		/// </summary>
+		/// <param name="other">The <see cref="VkExtent2D"/> to compare with this instance.</param>
+		/// <returns>
+		/// <c>true</c> if the specified <see cref="VkExtent2D"/> is equal to this instance; otherwise, <c>false</c>.
+		/// </returns>
+		public bool Equals(ref VkExtent2D other)
+		{
+			return other.width == width  && other.height == height;
```
Comment (mellinoe) [inline src/vk/VkExtent2D.cs]: nit: extra space before `&&`

File: src/vk/VkExtent2D.cs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+		/// <summary>
+		/// Determines whether the specified <see cref="object"/> is equal to this instance.
+		/// </summary>
+		/// <param name="obj">The <see cref="object"/> to compare with this instance.</param>
+		/// <returns>
+		/// <c>true</c> if the specified <see cref="object"/> is equal to this instance; otherwise, <c>false</c>.
+		/// </returns>
+		public override bool Equals(object obj)
+		{
+			return obj is VkExtent2D && Equals((VkExtent2D)obj);
```
Comment (mellinoe) [inline src/vk/VkExtent2D.cs]: nit, you can use the new "is" syntax and do:

```C#
return obj is VkExtent2D extent && Equals(extent)
```

File: src/vk/VkOffset2D.cs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+		/// <summary>
+		/// Determines whether the specified <see cref="object"/> is equal to this instance.
+		/// </summary>
+		/// <param name="obj">The <see cref="object"/> to compare with this instance.</param>
+		/// <returns>
+		/// <c>true</c> if the specified <see cref="object"/> is equal to this instance; otherwise, <c>false</c>.
+		/// </returns>
+		public override bool Equals(object obj)
+		{
+			return obj is VkOffset2D && Equals((VkOffset2D)obj);
```
Comment (mellinoe) [inline src/vk/VkOffset2D.cs]: Same comment about the "is" syntax.

File: src/vk/VkRect2D.cs [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="Rect2D"/> structure.
+		/// </summary>
+		/// <param name="x">The X component of the offset.</param>
+		/// <param name="y">The Y component of the offset.</param>
+		/// <param name="width">The width component of the extent.</param>
+		/// <param name="height">The height component of the extent.</param>
+		public VkRect2D(int x, int y, uint width, uint height)
+        {
```
Comment (mellinoe) [inline src/vk/VkRect2D.cs]: Spacing is wrong

File: src/vk/VkRect2D.cs [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+		/// <summary>
+		/// Determines whether the specified <see cref="object"/> is equal to this instance.
+		/// </summary>
+		/// <param name="obj">The <see cref="object"/> to compare with this instance.</param>
+		/// <returns>
+		/// <c>true</c> if the specified <see cref="object"/> is equal to this instance; otherwise, <c>false</c>.
+		/// </returns>
+		public override bool Equals(object obj)
+		{
+			return obj is VkRect2D && Equals((VkRect2D)obj);
```
Comment (mellinoe) [inline src/vk/VkRect2D.cs]: "is" syntax

**Comment by amerkoleci** [comment]
Can you add .editorconfig on root folder? So I can inherit your settings :)

**Comment by mellinoe** [comment]
> Can you add .editorconfig on root folder? So I can inherit your settings :)

It's a good idea -- I'll just have to figure out how to do it 😄

**Comment by amerkoleci** [comment]
Just pushed new .editorconfig file.

File: src/samples/texture/TextureMappingExample.cs [hunk 9]
```diff
@@ -272,11 +272,11 @@ void loadTexture(string fileName, VkFormat format, bool forceLinearTiling)
 
                 // Create optimal tiled target image
                 VkImageCreateInfo imageCreateInfo = Initializers.imageCreateInfo();
-                imageCreateInfo.imageType = VkImageType._2d;
+                imageCreateInfo.imageType = VkImageType.Image2D;
```
Comment (mellinoe) [inline src/samples/texture/TextureMappingExample.cs:275]: I like this set of changes. 👍

**Comment by amerkoleci** [comment]
Let me fix tabbing then you can merge

**Comment by amerkoleci** [comment]
Should be good now.

File: src/samples/texturearray/TextureArrayExample.cs [hunk 10]
```diff
@@ -192,7 +192,7 @@ void loadTextureArray(string filename, VkFormat format)
             imageCreateInfo.usage = VK_IMAGE_USAGE_SAMPLED_BIT;
             imageCreateInfo.sharingMode = VK_SHARING_MODE_EXCLUSIVE;
             imageCreateInfo.initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
-            imageCreateInfo.extent = { textureArray.width, textureArray.height, 1 };
+            imageCreateInfo.extent = new  { textureArray.width, textureArray.height, 1 };
```
Comment (mellinoe) [inline src/samples/texturearray/TextureArrayExample.cs:195]: By the way, I never actually finished this sample project. So don't worry if it fails to compile.

File: src/vk.generator/EnumHelpers.cs [hunk 11]
```diff
@@ -16,7 +16,27 @@ public static class EnumHelpers
 
         private static readonly Dictionary<string, string> s_knownEnumValueNames = new Dictionary<string, string>
         {
-            {  "VK_STENCIL_FRONT_AND_BACK", "FrontAndBack" }
+            {  "VK_STENCIL_FRONT_AND_BACK", "FrontAndBack" },
+			// VkSampleCountFlagBits
```
Comment (mellinoe) [inline src/vk.generator/EnumHelpers.cs:20]: There's still some tabs in here which cause the formatting to be wrong.

**Comment by amerkoleci** [comment]
Can you merge and maybe i can send another pull request

**Comment by mellinoe** [comment]
@amerkoleci I'll pull down your changes locally and do the final fixups before I merge this.

**Comment by mellinoe** [comment]
@amerkoleci I fixed up the remaining tabs, bumped the version to 1.0.3, and published a new nuget package.

**Pull request merged**